File: medium/2225_find_players_with_zero_or_one_losses/solution.cpp
#include <algorithm>
#include <set>
#include <vector>

class Solution {
public:
  std::vector<std::vector<int>>
  findWinners(std::vector<std::vector<int>> &matches) {
    std::unordered_map<int, int> lossCounter;

    for (const std::vector<int> &match : matches) {
      lossCounter[match[0]];
      ++lossCounter[match[1]];
    }

    std::vector<std::vector<int>> answer(2);

    for (const auto &[loser, count] : lossCounter) {
      if (count < 2)
        answer[count].push_back(loser);
    }

    sort(answer[0].begin(), answer[0].end());
    sort(answer[1].begin(), answer[1].end());

    return answer;
  }
};

File: medium/2225_find_players_with_zero_or_one_losses/solution.py
from collections import defaultdict
from typing import List


class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        loss_counter = defaultdict(int)

        for winner, loser in matches:
            loss_counter[winner]
            loss_counter[loser] += 1

        answer = [[], []]

        for loser in sorted(loss_counter.keys()):
            count = loss_counter[loser]

            if count < 2:
                answer[count].append(loser)

        return answer
